packages/sdk/src/types/VersionedModuleSource.ts

```diff
@@ -100,22 +99,50 @@ export async function parseModuleInterface(moduleSource: VersionedModuleSource):
  * Extract the embedded smart contract schema bytes. Returns `null` if no schema is embedded.
  * @param {VersionedModuleSource} moduleSource The smart contract module source.
  * @returns {RawModuleSchema | null} The raw module schema if found.
+ * @throws If the module source cannot be parsed or contains duplicate schema sections.
  */
-export async function getEmbeddedModuleSchema(moduleSource: VersionedModuleSource): Promise<RawModuleSchema | null> {
-    const wasmModule = await WebAssembly.compile(moduleSource.source);
-    const versionedSchema = schemaBytesFromWasmModule(wasmModule, 'concordium-schema');
-    if (versionedSchema !== null) {
-        return { type: 'versioned', buffer: versionedSchema };
+export function getEmbeddedModuleSchema({ source, version }: VersionedModuleSource): RawModuleSchema | undefined {
+    const sections = findCustomSections(new WebAssembly.Module(source), version);
```

Comment:
We should not use `WebAssembly.Module` since this will block the main thread and for large smart contracts this might freeze the UI.